pulp_smash/pulp3/pytest_plugin/__init__.py

```diff
@@ -0,0 +1,236 @@
+import asyncio
+import threading
+import socket
+import ssl
+
+from dataclasses import dataclass
+from typing import Any
+
+from aiohttp import web
+
+import pytest
+import trustme
+
+from pulp_smash.api import _get_sleep_time
+from pulp_smash.config import get_config
+from pulp_smash.pulp3.bindings import delete_orphans, monitor_task
+from pulp_smash.pulp3.fixture_utils import add_recording_route
+
+from pulpcore.client.pulpcore import ApiClient, TasksApi
+
+
+cfg = get_config()
+SLEEP_TIME = _get_sleep_time(cfg)
+
+
+class ThreadedAiohttpServer(threading.Thread):
+    def __init__(self, shutdown_event, app, host, port, ssl_ctx):
+        super().__init__()
+        self.shutdown_event = shutdown_event
+        self.app = app
+        self.host = host
+        self.port = port
+        self.ssl_ctx = ssl_ctx
+
+    def run(self):
+        loop = asyncio.new_event_loop()
+        runner = web.AppRunner(self.app)
+        asyncio.set_event_loop(loop)
+        loop.run_until_complete(runner.setup())
+        site = web.TCPSite(runner, host=self.host, port=self.port, ssl_context=self.ssl_ctx)
+        loop.run_until_complete(site.start())
+        while True:
+            loop.run_until_complete(asyncio.sleep(1))
```

Comment:
This is replacing the run_forever here?

Reply:
Yes it is. It's checking for the shutdown event and otherwise goes back to serving fixtures data.